### PR TITLE
Speed up Qwen3-VL-8B computer-use rollouts

### PR DESCRIPTION
**Pull request by zhouhelena1**
## Summary

Tunes the Qwen3-VL-8B GRPO tutorial (`rl/008_computer_use`) to speed up the 15-rollout demo from **30m10s → 18m55s** on the same 8×H100 node while preserving the reward curve.
 
All tuning is in the **tutorial's `_train()` overrides**. Left the shared `Qwen3_VL_8b_Recipe` unchanged, because the knobs are uniquely fitted to this tutorial's short coordinate outputs and would regress general use of the recipe. This was the logic:
> short outputs → want high SGLang rollout concurrency → `mem_fraction=0.75` → need `TP=4` to shard the 8B weights enough that the 0.75 KV pool fits without OOMing.
 
Where the speedup comes from (see [training gym dashboard](https://modal-labs-helena-dev--training-gym-dashboard-fastapi-app.modal.run/training) `acoustic-shell` → `espressivo-calico`):
 
- **Final checkpoint: 527s → 145s (−382s)** — `no_save_optim=True` + `save_interval=15` (skips the mid-run save at step 10 and the optimizer state)
- **Rollout steps: −194s** — mostly `rollout_max_response_len` 256 → 64.
 
Other things I tried: `TP=2`/`mem=0.55` with everything else at the optimized values (resp_len=64, save_interval=15, no_save_optim) ran **much slower** (30+ minutes) than `espressivo-calico` (`TP=4`/`mem=0.75`). And `TP=2`/`mem=0.75` OOMs.

image

Full wandb results here: https://wandb.ai/helenaaspamm/computer-use-grounding

## Checklist

- [x] Example is documented with comments throughout, in a [_Literate Programming_](https://en.wikipedia.org/wiki/Literate_programming) style.
- [x] Example does _not_ require third-party dependencies to be installed locally
- [x] Example pins its dependencies
  - [x] Example pins container images to a stable tag, not a dynamic tag like `latest`
  - [x] Example specifies a `python_version` for the base image, if it is used
  - [x] Example pins all dependencies to at least minor version, `~=x.y.z` or `==x.y`
  - [x] Example dependencies with `version < 1` are pinned to patch version, `==0.y.z`

---

<a href="https://modal.devinenterprise.com/review/modal-projects/training-gym/pull/259" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

File: modal_training_gym/train_recipes/slime_recipe/qwen3_vl_8b.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ class Qwen3_VL_8b_Recipe(SlimeRecipe):
 
     gpu_type: str = "H100"
     colocate: bool = True
-    tensor_model_parallel_size: int = 2
+    tensor_model_parallel_size: int = 4
```
[the remark this answers is not in the document]
Comment (zhouhelena1) [reply modal_training_gym/train_recipes/slime_recipe/qwen3_vl_8b.py]: Can confirm this works

**Comment by joyliu-q** [comment]
I found a prior deployment that runs faster thaan the speedup you made: https://github.com/modal-projects/training-gym/pull/254#issuecomment-4983468195

| Model | Status | Duration | Steps | Run |
| --- | --- | --- | --- | --- |
| Qwen3-VL-8B-Instruct | ✅ completed | 360.0s | 1 | [`every-union-9bdfbfa2ff07`](https://modal-labs-training-gym--training-gym-dashboard-fastapi-app.modal.run/training/every-union-9bdfbfa2ff07) | -112.0s (-24%) from [`affable-methodology-6277ce11c8eb`](https://modal-labs-training-gym--training-gym-dashboard-fastapi-app.modal.run/training/affable-methodology-6277ce11c8eb) |

compared to 

| Model | Status | Duration | Steps | Run |
| --- | --- | --- | --- | --- |
| Qwen3-VL-8B-Instruct | ✅ completed | 658.0s | 1 | `unsorted-snipe-cd3a78dc04b9` |

File: modal_training_gym/train_recipes/slime_recipe/qwen3_vl_8b.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (zhouhelena1) [reply modal_training_gym/train_recipes/slime_recipe/qwen3_vl_8b.py]: Not applicable now since changes removed from recipe

File: tutorials/rl/008_computer_use/008_computer_use.py [hunk 2]
```diff
@@ -341,14 +341,17 @@ def _main_impl() -> None:
         model=base_model,
         dataset=train_dataset,
         recipe=Qwen3_VL_8b_Recipe(
+            tensor_model_parallel_size=4,
```
Comment (joyliu-q) [inline tutorials/rl/008_computer_use/008_computer_use.py:351]: Can you add some comments on why you made these modifications, explaining to users how these speed things up so they can also do this themselves?

File: tutorials/rl/008_computer_use/008_computer_use.py [hunk 3]
```diff
@@ -341,14 +341,17 @@ def _main_impl() -> None:
...
             custom_rm_function=grounding_reward,
             num_rollout=15,
             rollout_batch_size=8,
             n_samples_per_prompt=4,
-            rollout_max_response_len=256,
+            rollout_max_response_len=64,
+            sglang_mem_fraction_static=0.75,
```
Comment (joyliu-q) [inline tutorials/rl/008_computer_use/008_computer_use.py:360]: Same here saying something like `# We only need 64 tokens here because the model is only outputting coordinates.`

File: tutorials/rl/008_computer_use/008_computer_use.py [hunk 4]
```diff
@@ -341,14 +341,17 @@ def _main_impl() -> None:
...
             global_batch_size=16,
             lr=1e-6,
-            save_interval=10,
+            save_interval=15,
```
Comment (joyliu-q) [inline tutorials/rl/008_computer_use/008_computer_use.py:363]: No need to comment here while bumping it

File: tutorials/rl/008_computer_use/008_computer_use.py [hunk 5]
```diff
@@ -341,14 +341,17 @@ def _main_impl() -> None:
...
+            no_save_optim=True,
```
Comment (joyliu-q) [inline tutorials/rl/008_computer_use/008_computer_use.py:366]: Add comment: what does this new flag do?

File: tutorials/rl/008_computer_use/008_computer_use.py (hunk 2 above)
Comment (zhouhelena1) [reply tutorials/rl/008_computer_use/008_computer_use.py:351]: Yes, just added!

File: tutorials/rl/008_computer_use/008_computer_use.py (hunk 3 above)
Comment (zhouhelena1) [reply tutorials/rl/008_computer_use/008_computer_use.py:360]: Added!

File: tutorials/rl/008_computer_use/008_computer_use.py (hunk 5 above)
Comment (zhouhelena1) [reply tutorials/rl/008_computer_use/008_computer_use.py:366]: Added!

**Review by joyliu-q — APPROVED** [review]
